### DISCUSSION TITLE
Highlighting Favorite Features in Rabbit

### DISCUSSION THREAD

**Discussion by kingdonb** [rabbit-shocker/rabbit#157] [Show and tell]
Hi! I've been learning my skill as a presenter for several years and as a user of Rabbit. Today, I'm excited to share my experience!

I've done several presentations where I started to learn Rabbit, by diving in and using it at conference talks! The killer feature that probably everyone knows first, is the rabbit and tortoise timer animation. If you ever saw anyone presenting with two icons chasing each other across the screen, to represent the time left in the presentation then it was probably Rabbit-Shocker! The timer is great for practicing talks, but if it's styled and built into the presentation, then we might as well use it for the big production show as well.

### Timer: Production Values

Great showmen like Matz can make you a little bit nervous with the humble brag about how many slides there are left, oh no! How will he make it to the end of the talk on time 😂 they can build their slides so that many content slides can be rolled through quickly, and a picture slowly builds up in the mind of the audience to fit together the different parts from the presenter's complex content.

Rabbit users can also build their own custom themes, so it does not always have to be the same tortoise and hare timer, to give the appearance of a talk your own personal brand. Rabbit can animate slides, which we do with some control of the order of animation!

Rabbit works entirely with markdown, and can fully integrate this animation and many other features into the plain-text format.

```markdown
= Features: UI (2)

  # wait
  * Index page
  * Graffiti
  * (('wait'))Pause
  * (('wait'))I18N
  * Search
```

I've used other slide tools that have similar animation feature, but it's always top-to-bottom and low fidelity of control. You cannot decide to leave a bullet blank in the middle of a bullet list and only show it later. This seems like an obvious but important feature.

If you want to slow reveal bullet points, in those old tools you had to do it from top to bottom, and all bullets will be slow-revealed in the same order every time. This gets very repetitive and lacks flexibility in structure, putting audiences to sleep. It also makes too many transitions, when I wanted one or two transitions on the slide, and it creates ugly slides with poorly planned positional structure, where the focal point delivered as a punchline is not able to be highlighted at the top when you want.

So you don't have any dimensionality without this capability. In Rabbit, you can control the order of display and reduce the number of transitions necessary to achieve the desired effect, to obtain a boost to the engagement with your audience!

#### Pausing for Engagement

Sometimes I want to show more structure and make the audience anticipate hearing about what's next, we can show a fill-in the blank. Tell a few things, but leave one for the audience member to guess "does anyone know this..." They will pay attention to what you are saying if you don't put text about so-far irrelevant points or topics not being discussed yet for them to read ahead of time.

This is a great engagement hack! Show only the smallest amount of text that is important to what you are about to say.

With this added control over flow, once you have used it, I feel it's very hard to go back to old tools. It's great Rabbit can do that and other basic formats without any widgets or formatting controls that have to live outside of the text file. Now we programmers can always use the same tools (Git!) to manage our slide content as we use for managing our codebases.

The best feature, my favorite feature about Rabbit slides is there is no formatting when you start writing, only content. You can focus on just writing, look at just words, and try not to put too much text on the slides while getting the point across :) I often got hung up on making slides look good with more common WYSIWYG tools that everybody uses. It is a bottomless pit to spend time. It's never perfect, only when we finally spent enough time on the look that we can't justify spending more time improving further.

In Rabbit, the style is separated from the content. You are not tempted to work on style and content at the same time, since style is programmed and content is in plain text (markdown.) This distinction between those activities is unbreakable.

### Theming Rabbit and Style

And yeah, we should spend the time to make our slides look good, but also spend the time on formatting in the diagrams where it matters most. Besides embedding images, we can use a nice tool like mermaid or charty. These are all supported in Rabbit and they are able to be used if you can install them on your system. Some need some dependencies to be pre-installed.

The new Mermaid diagram feature downloads most of the JavaScript dependencies automatically, you just need nodejs. It produces a PDF cache so diagrams that didn't change since you practiced the slides won't have to re-render again.

Rabbit users can change the slides content in a text file live while the presentation is running and see Rabbit update live, it's nice!

#### Live Iterative Editing

The defaults in most slide themes from other tools IMHO aren't very good. They typically promote use of smaller fonts by default, and I will usually want to turn up the font size to avoid having small text, unless I am putting way too much text on my slides. On Rabbit, the default font size is large so I don't get caught thinking about this. I just keep one main idea per slide, and make more slides when the text of the idea grows beyond the slide text limit. This is more productive than resizing the fonts to fit more text.

By the time I had to adjust the font size, I am already obsessing about other details like image placement, font selection, and probably kerning too, IDK, all these things are rabbit-holes. (¡Hah!!!) Rabbit saves me from all of this.

It's so helpful to separate this content from style, but Rabbit can do it without under-emphasizing style itself. We can make an elaborate theme and spend time to make sure it looks good, then just use it later without thinking too much about it.

I am very grateful to the authors of Rabbit (and Ruby!)

### Wrap-up

I am glad to find the Rabbit discussion board, I hope to meet more Rabbit users and I do hope if you see me in a public space and you recognize me, you will bring up Rabbit and Ruby, or tell me something about your presentation style 🚀🤠 my name is Kingdon

😎 I think that most presenters will prefer to talk about the actual abstracted topic of their talk rather than the structure and design of their slides, but honestly I wouldn't put it past us, many of us spend a lot of energy to make sure it looks just right.

I will not mind at all if you ask me about my slides, and I appreciate every time whenever someone notices these details.

**Comment by kingdonb** [comment]
I almost forgot, here are my latest slides! (I need to do a "compiled works of self" cv - still don't have anything like that!)

https://github.com/kingdonb/dot-s2e2-rabbit-slides

I learned a lot from talking with @kou on the bug tracker. You can embed a full terminal in Rabbit! Holy 🐮 

Now I will remember these features for next time, I wrote down the advanced features, in the video slides I forgot to use `wait` pause feature and some other features in the last talk! (It's great that Ruby is always open source, so it's not hard to see how features work!)

**Comment by kou** [comment]
Thanks for writing this!
This will useful for all other Rabbit users (Rabbit Shocker)!

BTW, you may want to share your slide on https://slide.rabbit-shocker.org/ . You can just push your slide as a gem to share your slide on https://sldie.rabbit-shocker.org/ . See also: https://rabbit-shocker.org/en/usage/rabbit-slide.html
